sadu-queries2/src/main/java/de/chojo/sadu/queries/configuration/QueryConfiguration.java

```diff
@@ -0,0 +1,179 @@
+        return new QueryConfigurationBuilder(source);
+    }
+
+    /**
+     * Returns a new QueryConfiguration object with the provided query and other configuration settings.
+     *
+     * @param query the query to be associated with the configuration
+     * @return the new QueryConfiguration object
+     */
+    public QueryConfiguration forQuery(QueryImpl query) {
+        return new QueryConfiguration(query, dataSource, atomic, throwExceptions, exceptionHandler, rowMapperRegistry);
+    }
+
+    /**
+     * Handles a SQLException by invoking the exceptionHandler consumer, logging the exception,
+     * and potentially throwing a WrappedQueryExecutionException.
+     *
+     * @param e the SQLException to handle
+     */
+    public void handleException(SQLException e) {
+        exceptionHandler.accept(e);
+        query.logException(e);
+        if (throwExceptions) {
+            throw (WrappedQueryExecutionException) new WrappedQueryExecutionException(e.getMessage()).initCause(e);
+        }
+    }
+
+    /**
+     * Retrieves the value of the atomic flag.
+     *
+     * @return {@code true} if atomic flag is set; {@code false} otherwise.
+     */
+    public boolean atomic() {
+        return atomic;
+    }
+
+    /**
+     * Retrieves the value of the throwExceptions field.
+     *
+     * @return the value of the throwExceptions field
+     */
+    public boolean throwExceptions() {
+        return throwExceptions;
+    }
+
+    /**
+     * Retrieves the {@link RowMapperRegistry} object from the {@link QueryConfiguration}.
+     *
+     * @return The {@link RowMapperRegistry} object from the {@link QueryConfiguration}.
+     */
+    public RowMapperRegistry rowMapperRegistry() {
+        return rowMapperRegistry;
+    }
+
+    /**
+     * Returns the DataSource object associated with this QueryConfiguration.
+     *
+     * @return the DataSource object
+     */
+    public DataSource dataSource() {
+        return dataSource;
+    }
+
+    /**
+     * Throws an IllegalStateException with a specific error message.
+     * This method is meant to be called when the QueryConfiguration doesn't hold a configuration.
+     * To create a connection, use the {@link #withSingleTransaction} method and make sure to use try-with-resources on the new configuration instance.
+     *
+     * @throws IllegalStateException if the QueryConfiguration doesn't hold a configuration
+     */
+    public Connection connection() {
+        throw new IllegalStateException("Query configuration doesnt hold a configuration.\nCall #withSingleTransaction for a connection to be created and use try with resources on the new configuration instance");
+    }
+
+    /**
+     * Checks if the current query configuration has an active connection.
+     *
+     * @return {@code true} if there is an active connection, {@code false} otherwise.
+     */
+    public boolean hasConnection() {
+        return false;
```

Comment:
Its wrong based on the documentation

Reply:
By definition the QueryConfiguration never holds a connection, but the ConnectedQueryConfiguration can. Instead of checking here I just take a shortcut because the value will always be null.

Reply:
From my point of view, this can confuse beginners with the Lib. That is why I have added this point

Reply:
Yes it could, I will replace it with an instance check instead.